src/zenml/models/pipeline_run_models.py

```diff
@@ -212,19 +220,39 @@ def generate_filter(
             The filter expression for the query.
         """
         from sqlalchemy import and_
+        from sqlmodel import or_
 
         base_filter = super().generate_filter(table)
 
+        operator = (
+            or_ if self.logical_operator == LogicalOperators.OR else and_
+        )
+
         if self.unlisted is not None:
             if self.unlisted is True:
                 unlisted_filter = getattr(table, "pipeline_id").is_(None)
             else:
                 unlisted_filter = getattr(table, "pipeline_id").is_not(None)
 
-            # TODO: make this right
-            # This needs to be an AND right now to work with the workspace
-            # scoping of the superclass
-            return and_(base_filter, unlisted_filter)
+            base_filter = operator(base_filter, unlisted_filter)
+
+        if self.code_repository_id:
+            from zenml.zen_stores.schemas import (
+                CodeReferenceSchema,
+                PipelineDeploymentSchema,
+                PipelineRunSchema,
+            )
+
+            code_repo_filter = and_(
+                True,
```

Comment:
Why is it necessary to explicitely set the first Value to true here?

Reply:
That was a try for mypy, but didn't fix the issue either so I just added an ignore